### PR TITLE
364 cardinality many duplicate datoms

### PR DESCRIPTION
**Pull request by grischoun**
Fix the duplicate issue when inserting on cardinality many attributes.

### CODE REVIEW

**Comment by grischoun** [comment]
I still need to check why two tests fail.

**Comment by grischoun** [comment]
This is now ready for review.

There is potential for improving 'retraction' performance, I think. I'll investigate more on that later.

File: src/datahike/index/hitchhiker_tree/insert.cljc [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    (if (exists-old? kvs key)
+      kvs
+      (assoc kvs key nil)))
+  (-apply-op-to-tree [_ tree]
+    (let [children (cond
+                     (tree/data-node? tree) (:children tree)
+                     :else (:children (peek (tree/lookup-path tree key))))]
+      (if (exists-old? children key)
+        tree
+        (tree/insert tree key nil)))))
```
Comment (jsmassa) [inline src/datahike/index/hitchhiker_tree/insert.cljc:47]: How does this work so there are no duplicates when the key contains the transaction number? 
Don't you test for datoms from 2 different transactions? So how could the old key be found? 
I assume I am overlooking something here...

Comment (grischoun) [reply src/datahike/index/hitchhiker_tree/insert.cljc]: Good catch @jsmassa !
It now replaces the old key with the new one (like with the persistent-set index).

**Review by jsmassa — APPROVED** [review]
Nice, then I am happy now with the code :)

**Pull request merged**